### DISCUSSION TITLE
Does caffeine loading cache do automatic interning of cached values?

### DISCUSSION THREAD

**Discussion by pugupta-linkedin** [ben-manes/caffeine#809] [Q&A] [answered]
Hi we're creating a LoadingCache with caffeine with - (1) maxSize (2) variable expiration w/ its builder.
Can caffeine intern cached values automatically (i ask this since i'm debugging an issue and want to eliminate this as a root cause).
At the moment we don't use external interning mechanisms for either keys or values - also the cacheloader always creates new objects (never reuses or returns static objects).

cc @ben-manes

**Comment by ben-manes** [comment, marked as answer]
No, nothing magical like that. You can think of it as a `ConcurrentHashMap` with an automatic removal policy. The entries are shared until evicted, loaded if not found, etc. It does not try to keep them around any longer, e.g. by calling `String.intern`. It doesn't copy objects into or out of the map, etc. It should be boring and feel similar to the core Java collections.

(We do offer an [Interner](https://github.com/ben-manes/caffeine/wiki/Interner) for those who want that behavior)

**Reply by pugupta-linkedin** [reply]
Thanks for the answer!
